sherlock.h

```diff
@@ -115,34 +115,47 @@ class PubSubHTTPEndpoint {
       : value_name_(value_name),
         http_request_(std::move(r)),
         http_response_(http_request_.SendChunkedResponse()) {
-    // TODO(dkorolev): `"from"`, etc.
     if (http_request_.url.query.has("recent")) {
-      serving_ = false;
+      serving_ = false;  // Start in 'non-serving' mode when `recent` is set.
       from_timestamp_ =
           r.timestamp - static_cast<bricks::time::MILLISECONDS_INTERVAL>(
                             bricks::strings::FromString<uint64_t>(http_request_.url.query["recent"]));
-    } else {
-      serving_ = true;
     }
-    n_ = http_request_.url.query.has("n") ? bricks::strings::FromString<size_t>(http_request_.url.query["n"])
-                                          : 0;
+    if (http_request_.url.query.has("n")) {
+      serving_ = false;  // Start in 'non-serving' mode when `n` is set.
+      bricks::strings::FromString(http_request_.url.query["n"], n_);
+      cap_ = n_;  // If `?n=` parameter is set, it sets `cap_` too by default. Use `?n=...&cap=0` to override.
+    }
+    if (http_request_.url.query.has("n_min")) {
+      // `n_min` is same as `n`, but it does not set the cap; just the lower bound for `recent`.
+      serving_ = false;  // Start in 'non-serving' mode when `n_min` is set.
+      bricks::strings::FromString(http_request_.url.query["n_min"], n_);
+    }
+    if (http_request_.url.query.has("cap")) {
+      bricks::strings::FromString(http_request_.url.query["cap"], cap_);
+    }
   }
 
-  inline bool Entry(E& entry) {
+  inline bool Entry(E& entry, size_t index, size_t total) {
     // TODO(dkorolev): Should we always extract the timestamp and throw an exception if there is a mismatch?
     try {
       if (!serving_) {
-        // The only condition we have now to not serve the entries via HTTP is the `from` URL parameter.
         const bricks::time::EPOCH_MILLISECONDS timestamp = ExtractTimestamp(entry);
-        if (timestamp >= from_timestamp_) {
+        // Respect `n`.
+        if (total - index <= n_) {
+          serving_ = true;
+        }
+        // Respect `recent`.
+        if (from_timestamp_ != static_cast<bricks::time::EPOCH_MILLISECONDS>(-1) &&
```

Comment:
I thought of it while typing. It would have to go into Bricks though, and I wasn't ready for a cross-repository change.